### PR TITLE
Remove pytorch warning message

### PR DESCRIPTION
**Pull request by yifan0330**
Closes # .

Changes proposed in this pull request:

- Replace `functorch.hessian` by `torch.func.hessian` in cbmr code, to remove warning message if pytorch version >=2.0.

### CODE REVIEW

File: nimare/diagnostics.py [hunk 1]
```diff
@@ -466,23 +466,18 @@ def transform(self, dataset):
...
 
-        # Check if each coordinate in Dataset is within the mask
-        # If it is, log that coordinate in keep_idx
-        keep_idx = [
-            i
-            for i, coord in enumerate(dset_ijk)
-            if len(np.where((mask_ijk == coord).all(axis=1))[0])
-        ]
+        # Only retain coordinates inside the brain mask
+        keep_idx = [i for i, coord in enumerate(dset_ijk) if masker_array[coord[0], coord[1], coord[2]] == 1]
```
Comment (jdkent) [inline nimare/diagnostics.py]: nice! this is several orders of magnitude faster! :rocket:

File: nimare/meta/cbmr.py [hunk 2]
```diff
@@ -110,10 +110,10 @@ def __init__(
         spline_spacing=10,
         model=models.PoissonEstimator,
         penalty=False,
-        n_iter=1000,
-        lr=1e-2,
+        n_iter=2000,
+        lr=1,
         lr_decay=0.999,
```
Comment (jdkent) [inline nimare/meta/cbmr.py:115]: what's the rationale for the changes to the defaults? is it because of the changes to the optimization procedure?

File: nimare/meta/models.py [hunk 3]
```diff
@@ -45,10 +43,10 @@ def __init__(
         spatial_coef_dim=None,
         moderators_coef_dim=None,
         penalty=False,
-        lr=0.1,
+        lr=1,
         lr_decay=0.999,
-        n_iter=1000,
-        tol=1e-2,
+        n_iter=2000,
+        tol=1e-9,
```
Comment (jdkent) [inline nimare/meta/models.py:49]: same here, how were defaults decided?

File: nimare/meta/cbmr.py (hunk 2 above)
Comment (jdkent) [reply nimare/meta/cbmr.py:115]: the rationale is to take a full step for the initial iteration, then the learning rate decay will exponentially reduce the step size.

**Review by jdkent — APPROVED** [review]
LGTM!
